File: app/src/main/java/com/dammaj/app/data/ScraperCallback.java
package com.dammaj.app.data;

import com.dammaj.app.model.ContentItem;
import com.dammaj.app.model.AudioItem;

import java.util.List;

public interface ScraperCallback {
    void onSuccess(List<ContentItem> contentItems);
    void onFailure(Exception e);
    
    // For audio content
    void onAudioSuccess(List<AudioItem> audioItems);
    void onAudioFailure(Exception e);
}

File: app/src/main/java/com/alilmia/app/data/ScraperCallback.kt
package com.alilmia.app.data

import com.alilmia.app.model.ContentItem

/**
 * Callback interface for web scraping operations
 */
interface ScraperCallback {
    /**
     * Called when scraping succeeds
     * @param items List of scraped content items
     */
    fun onSuccess(items: List<ContentItem>)
    
    /**
     * Called when scraping fails
     * @param error Exception that occurred during scraping
     */
    fun onFailure(error: Exception)
}